WcaOnRails/app/assets/javascripts/competitions.js

```diff
@@ -52,6 +52,10 @@ onPage('competitions#edit, competitions#update, competitions#admin_edit, competi
     $('.adjacent-competitions .wca-adjacent-competitions-hide-events').show();
   });
 
+  $('select[name="competition[guest_entry_status]"]').on('change', function() {
+    $('.competition_guests_per_registration_limit').toggle(this.value === 'restricted');
```

Comment:
I don't think there's a way to access the enum from here, but please let me know if I'm missing something @gregorbg.

Reply:
I think in theory you could move the entire thing over to the `html.erb` world and inject a Ruby `<%= %>` template there. But it's really not necessary since we're looking to migrate the entire form to React soon anyways.